### PR TITLE
Format release notes from JSON

### PR DESCRIPTION
**Pull request by brianstrauch**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----
Take the JSON representation of the release notes from confluentinc/cli-release, and format for:
* GitHub
* S3
* Docs
* More formats to come, such as Homebrew, APT, etc.

Test & Review
-------------
Unit tested, ran make targets with `DRY_RUN=true`

### CODE REVIEW

File: cmd/releasenotes/main.go [hunk 1]
```diff
@@ -1,20 +1,124 @@
...
+		fmt.Print(docs(r))
+	case "github":
+		fmt.Print(github(r))
+	case "s3":
+		fmt.Print(s3(r))
+	}
+}
+
+func docs(r *releaseNotes) string {
+	title := fmt.Sprintf("[%s] Confluent CLI v%s Release Notes", time.Now().Format("1/2/2006"), "3.10.0")
```
Comment (sgagniere) [inline cmd/releasenotes/main.go:56]:
```suggestion
	title := fmt.Sprintf("[%s] Confluent CLI v%s Release Notes", time.Now().Format("1/2/2006"), r.version)
```

File: cmd/releasenotes/main.go [hunk 2]
```diff
@@ -1,20 +1,124 @@
...
+			out += bullet(line) + "\n"
+		}
+		out += "\n"
+	}
+
+	return out
+}
+
+func s3(r *releaseNotes) string {
+	title := fmt.Sprintf("[%s] Confluent CLI v%s Release Notes", time.Now().Format("1/2/2006"), "3.10.0")
```
Comment (sgagniere) [inline cmd/releasenotes/main.go:96]:
```suggestion
	title := fmt.Sprintf("[%s] Confluent CLI v%s Release Notes", time.Now().Format("1/2/2006"), r.version)
```

**Review by sgagniere — CHANGES_REQUESTED** [review]

File: cmd/releasenotes/main.go [hunk 3]
```diff
@@ -1,20 +1,124 @@
...
+	file, err := os.Open(path)
+	if err != nil {
 		panic(err)
 	}
 
-	if err := r.Write(releaseNotesPath); err != nil {
+	version := strings.TrimSuffix(filepath.Base(path), filepath.Ext(path))
+
+	r := &releaseNotes{
+		date:     time.Now(),
```
Comment (sgagniere) [inline cmd/releasenotes/main.go:36]: This field doesn't seem like it's used. In the `docs` and `s3` functions you're calling `time.Now()`, so I think either this field should be removed, or those functions should use this.

**Review by sgagniere — APPROVED** [review]

File: cmd/releasenotes/main.go [hunk 4]
```diff
@@ -1,20 +1,124 @@
...
+	"Bug Fixes",
+}
 
+// Usage: go run main.go 3.10.0.json docs
 func main() {
-	r := releasenotes.New()
+	path, format := os.Args[1], os.Args[2]
 
-	if err := r.ReadFromGithub(); err != nil {
+	file, err := os.Open(path)
```
Comment (MuweiHe) [inline cmd/releasenotes/main.go:28]: does it need a `defer close()`

Comment (brianstrauch) [reply cmd/releasenotes/main.go:28]: Doesn't hurt, thanks!

**Pull request merged**